### ISSUE TITLE
Error message missing if Http path part is not provided

### ISSUE THREAD

**Issue by warpech** [Starcounter/Starcounter#2710]
Assume the following project structure:

```txt
wwwroot/PersonPage.html
PersonPage.json
PersonPage.json.cs
Program.cs
```

If, in my `Program.cs` I provide the path to the HTML file without the `/` at the beginning:

```c#
Handle.GET ("/myapp", () => {
    PersonPage page = new PersonPage() {
        Html = "PersonPage.html"
    };
    return page;
});
```

The browser gets "406 OK" response with no error message at all. Also the Starcounter Log does not show an error message.

This works ok:

```c#
Handle.GET ("/myapp", () => {
    PersonPage page = new PersonPage() {
        Html = "/PersonPage.html"
    };
    return page;
});
```

Expected: There should be an error saying that the path to HTML file needs to begin with `/`. Also, the correct description for 406 is "Not acceptable", not "OK".

**Comment by miyconst** [comment]
+1

I was going to create this issue, but for me it doesn't response at all.

**Comment by alemoi** [comment]
I think I can take this one.

**Comment by alemoi** [comment]
> There should be an error saying that the path to HTML file needs to begin with /

I tried to implement this via check in Page.cs Html property like this: 
```c#
        public string Html {
            get { return Template.Html.Getter(this); }
            set {
                // Checking that Html value starts with slash.
                if ((value != null) && (value.Length > 0) && (value[0] != '/')) {
                    throw new ArgumentOutOfRangeException("Page Html property should start with a slash resembling a resource URI.");
                }

                Template.Html.Setter(this, value); }
        }
```
But it does not help because the `Html` property is not virtual so, for example, Launcher's `Html` property is totally independent, so the following code throws no error:

```c#
                    launcher = new Launcher() {
                        Html = "Launcher/LauncherTemplate.html"
                    };

```

Should `Html` from Page be virtual? @chrhol what do you think?

I will check now regarding "406 OK" case.

**Comment by warpech** [comment]
PS. I don't know if it matters, but the value of the `Html` property can come either from CS or from JSON.

**Comment by alemoi** [comment]
Ok, the check code for `page.Html` and fix for "406 Not acceptable" pushed via 9ec95c657dc5053b1f685a8b71efd58f360df812

Since there are many ways to set Html property, we need to discuss further.

**Comment by warpech** [comment]
I think this can be closed. The objective in OP was achieved. Please reopen if you disagree
